libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -96,50 +90,34 @@ void GCS_MAVLINK::handle_file_transfer_protocol(const mavlink_message_t &msg) {
     }
 }
 
-void GCS_MAVLINK::send_ftp_replies(void)
+bool GCS_MAVLINK::send_ftp_reply(const pending_ftp &reply)
 {
     /*
       provide same banner we would give with old param download
     */
-    if (ftp.need_banner_send_mask & (1U<<chan)) {
-        ftp.need_banner_send_mask &= ~(1U<<chan);
+    if (ftp.need_banner_send_mask & (1U<<reply.chan)) {
+        ftp.need_banner_send_mask &= ~(1U<<reply.chan);
         send_banner();
     }
-    
-    if (ftp.replies == nullptr || ftp.replies->is_empty()) {
-        return;
-    }
-
-    for (uint8_t i = 0; i < 20; i++) {
-        if (!HAVE_PAYLOAD_SPACE(chan, FILE_TRANSFER_PROTOCOL)) {
-            return;
-        }
-        if ((i > 0) && comm_get_txspace(chan) < (2 * (packet_overhead() + MAVLINK_MSG_ID_FILE_TRANSFER_PROTOCOL_LEN))) {
-            // if this isn't the first packet we have to leave deadspace for the next message
-            return;
-        }
-
-        struct pending_ftp reply;
-        uint8_t payload[251] = {};
-        if (ftp.replies->peek(reply) && (reply.chan == chan)) {
-            put_le16_ptr(payload, reply.seq_number);
-            payload[2] = reply.session;
-            payload[3] = static_cast<uint8_t>(reply.opcode);
-            payload[4] = reply.size;
-            payload[5] = static_cast<uint8_t>(reply.req_opcode);
-            payload[6] = reply.burst_complete ? 1 : 0;
-            put_le32_ptr(&payload[8], reply.offset);
-            memcpy(&payload[12], reply.data, sizeof(reply.data));
-            mavlink_msg_file_transfer_protocol_send(
-                reply.chan,
-                0, reply.sysid, reply.compid,
-                payload);
-            ftp.replies->pop();
-            ftp.last_send_ms = AP_HAL::millis();
-        } else {
-            return;
-        }
+    WITH_SEMAPHORE(comm_chan_lock(reply.chan));
+    if (!HAVE_PAYLOAD_SPACE(chan, FILE_TRANSFER_PROTOCOL)) {
```

Comment:
What made the code and comment about needing space for a second packet no longer relevant?  Admittedly, I've no idea why that requirement existed...

Reply:
I think it was a bad attempt to try to limit bandwidth usage, so not needed now